### PR TITLE
Depth Buffer

### PR DESCRIPTION
**Pull request by Lokathor**
* [x] Make it work
* [x] Explain the working
* [x] Note that `DepthImage` would also check for CPU_VISIBLE in other programs, depending on how you wanted to use it and such.
* [x] Pass at least one (1) review.

### CODE REVIEW

File: book/src/09_depth_buffer.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+          store: AttachmentStoreOp::DontCare,
+        },
+        stencil_ops: AttachmentOps::DONT_CARE,
+        layouts: Layout::Undefined..Layout::DepthStencilAttachmentOptimal,
+      };
+```
+
+This is _similar to_ the color attachment.
+
+* With the depth attachment we can specify our own format since we don't have to
```
[the remark this answers is not in the document]
Comment (Lokathor) [reply book/src/09_depth_buffer.md]: 🌊

File: book/src/09_depth_buffer.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            &[subpass],
+            &[in_dependency, out_dependency],
+          )
+          .map_err(|_| "Couldn't create a render pass!")?
+      }
+```
+
+## Update the `depth_stencil`
+
+Where we have our descriptor sets and all that we need a new `depth_stencil` definition.
```
[the remark this answers is not in the document]
Comment (Lokathor) [reply book/src/09_depth_buffer.md]: but we've had a `depth_stencil` for several lessons now >_>

[the remark this answers is not in the document]
Comment (termhn) [reply book/src/09_depth_buffer.md]: ... right, lol

File: book/src/09_depth_buffer.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+      let memory_type_id = adapter
+        .physical_device
+        .memory_properties()
+        .memory_types
+        .iter()
+        .enumerate()
+        .find(|&(id, memory_type)| {
+          // BIG NOTE: THIS IS DEVICE LOCAL NOT CPU VISIBLE
+          requirements.type_mask & (1 << id) != 0
+            && memory_type.properties.contains(Properties::DEVICE_LOCAL)
```
Comment (aleksijuvani) [inline book/src/09_depth_buffer.md]: AMD exposes a memory type with both `DEVICE_LOCAL` and `CPU_VISIBLE` so ideally we would also check for `&& !memory_type.properties.contains(Properties::CPU_VISIBLE)` (per https://gpuopen.com/vulkan-device-memory/).

Comment (aleksijuvani) [reply book/src/09_depth_buffer.md]: Everything is both `DEVICE_LOCAL` and `CPU_VISIBLE` on Intel however, so perhaps it's not worth making the distinction here. Just need to be mindful of the optimal memory type for each vendor.

Comment (termhn) [reply book/src/09_depth_buffer.md]: as long as it is not exclusively cpu visible it should be fine afaik

Comment (Lokathor) [reply book/src/09_depth_buffer.md]: I can add a note about it, no problem

**Pull request merged**